modules/xmpp/XmppConnection.js

```diff
@@ -432,23 +475,53 @@ export default class XmppConnection extends Listenable {
         const { shard, websocketKeepAliveUrl } = this._options;
         const url = websocketKeepAliveUrl ? websocketKeepAliveUrl
             : this.service.replace('wss://', 'https://').replace('ws://', 'http://');
-        
-        return fetch(url)
+
+        // #bloomberg #shard @araje Pass request headers for debugging
+        const traceId = uuidv4();
+
+        // Debugging info for shard detection
+        logger.info(`[REQUEST ${traceId}] GET ${url}`);
+
+        const decodedToken = this._decodeKeepAliveAndCheckShardToken(url);
+
+        const requestHeaders = {
+            'Content-Type': 'application/json',
+            'X-Trace-Id': traceId,
+            'X-User-Id': decodedToken ? decodedToken.uuid : 'NO_UUID'
+        };
+
+        // #end
+
+        // #bloomberg #shard @araje Pass request headers for debugging
```

Comment:
Do we need a separate flag for this change?

Reply:
Do you mean a feature flag in config.js?  The intention of this change is to add some debugging information to the shard detection request that we always want to keep for troubleshooting connection instability. A feature flag could be set up to turn this debug trace off but I wanted to avoid introducing more complexity. The Prosody module handling the shard detection request does not require these request headers and will work regardless.

Reply:
I mean  we have 2 `// #bloomberg #shard @araje Pass request headers for debugging` one after another